cmd/list.go

```diff
@@ -52,19 +57,63 @@ func runList(opts *dobiOptions, listOpts listOptions) error {
 		return err
 	}
 
-	resources := filterResources(conf, listOpts)
-	descriptions := getDescriptions(resources)
+	tags := getTags(conf.Resources)
+	var descriptions []string
+	if listOpts.groups {
+		resources := filterResourcesTags(conf, listOpts)
+		descriptions = getDescriptionsByTag(resources)
+	} else {
+		resources := filterResources(conf, listOpts)
+		descriptions = getDescriptions(resources)
+	}
+
 	if len(descriptions) == 0 {
 		logging.Log.Warn("No resources found. Try --all or --tags.")
 		return nil
 	}
-
-	tags := getTags(conf.Resources)
-
 	fmt.Print(format(descriptions, tags))
 	return nil
 }
 
+func filterResourcesTags(conf *config.Config, listOpts listOptions) []resourceGroup {
+	tags := []resourceGroup{}
+	if listOpts.all {
+		// Add 'none' tag group accessible via [0] for resources with no tags configured
+		tags = append(tags, resourceGroup{tag: "none"})
+	}
+	for _, name := range conf.Sorted() {
+		res := conf.Resources[name]
+		if len(res.CategoryTags()) > 0 {
+			for _, tagname := range res.CategoryTags() {
+				currentGroupIndex := 0
+				if i, found := findGroup(tags, tagname); found {
```

Comment:
I guess the performance of this code is not super important, but normally for this kind of lookup I would create a map so that the lookup is `O(1)`, and at the end convert it back into a slice.  If this is working I'm ok to leave it as-is.

Reply:
Yes, its working that way. I'm also okay leaving it as-is.